src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingManager.cs

```diff
@@ -89,8 +89,7 @@ internal class PersistentStreamPullingManager : SystemTarget, IPersistentStreamP
             // Remove cast once we cleanup
             queueAdapter = qAdapter.Value;
             await this.queueBalancer.Initialize(this.streamProviderName, this.adapterFactory.GetStreamQueueMapper(), config.SiloMaturityPeriod, this.providerConfig);
-            var meAsQueueBalanceListener = this.AsReference<IStreamQueueBalanceListener>();
-            queueBalancer.SubscribeToQueueDistributionChangeEvents(meAsQueueBalanceListener);
+            queueBalancer.SubscribeToQueueDistributionChangeEvents(this);
```

Comment:
Why changing this? It used to be that the notification here were delivered as grain calls on this system target, which is the right way for system targets to communicate. There are some other examples when system targets call functions directly (breaking isolation), but this was not the case here. In this case it was done the right way.
So why doing it the wrong way now - call directly and then manually switch context?